app/models/orchestration_template_azure.rb

```diff
@@ -0,0 +1,88 @@
+class OrchestrationTemplateAzure < OrchestrationTemplate
+  def parameter_groups
+    # Azure format does not have the concept of parameter group
+    # Place all parameters in one group
+    [OrchestrationTemplate::OrchestrationParameterGroup.new(
+      :label      => "Parameters",
+      :parameters => parameters
+    )]
+  end
+
+  def parameters
+    raw_parameters = JSON.load(content)["parameters"]
+    (raw_parameters || {}).collect do |key, val|
+      parameter = OrchestrationTemplate::OrchestrationParameter.new(
+        :name          => key,
+        :label         => key.titleize,
+        :data_type     => val['type'],
+        :default_value => val['defaultValue'],
+        :hidden        => val['type'] == 'securestring'
+      )
+
+      add_metadata(parameter, val['metadata'])
+      add_allowed_values(parameter, val['allowedValues'])
+
+      parameter
+    end
+  end
+
+  def self.eligible_manager_types
+    [ManageIQ::Providers::Microsoft::CloudManager]
+  end
+
+  # return the parsing error message if not valid JSON; otherwise nil
```

Comment:
I find it really weird that a validate method returns nil (or anything falsey) when everything is ok.